tools/validators/ontology_validator/yamlformat/tests/unit_lib_test.py

```diff
@@ -170,26 +173,74 @@ def testUnitFolderAddFromConfigTooManyStandardUnits(self):
                           findings_lib.StandardUnitCountError)
 
   def testUnitWithIllegalKeyTypeHasFindings(self):
-    unit = unit_lib.Unit(False, 'invalid')
+    unit = unit_lib.Unit(False)
     self.assertIsInstance(unit.GetFindings()[0],
                           findings_lib.IllegalKeyTypeError)
 
   def testUnitWithIllegalNameHasFindings(self):
-    unit = unit_lib.Unit('BADUNIT', 'invalid')
+    unit = unit_lib.Unit('BADUNIT')
     self.assertIsInstance(unit.GetFindings()[0],
                           findings_lib.InvalidUnitNameError)
 
   def testUnitEquals(self):
-    unit_one = unit_lib.Unit('unit_one', 'measurement')
-    unit_one_dup = unit_lib.Unit('unit_one', 'measurement')
-    unit_one_diff_measurement = unit_lib.Unit('unit_one', 'changed')
-    unit_one_standard = unit_lib.Unit('unit_one', 'measurement', True)
-    unit_two = unit_lib.Unit('unit_two', 'measurement')
+    unit_one = unit_lib.Unit('unit_one')
+    unit_one_dup = unit_lib.Unit('unit_one')
+    unit_one_standard = unit_lib.Unit('unit_one', True)
+    unit_two = unit_lib.Unit('unit_two')
     self.assertEqual(unit_one, unit_one_dup)
-    self.assertNotEqual(unit_one, unit_one_diff_measurement)
     self.assertNotEqual(unit_one, unit_one_standard)
     self.assertNotEqual(unit_one, unit_two)
 
+  def testUnitFolderAddFromConfigMeasurementAlias(self):
+    doc = {
+      'length': 'distance',
+      'distance': [
+        {'meters': 'STANDARD'},
+        'feet'
+      ],
+    }
+    folder = unit_lib.UnitFolder(_GOOD_PATH)
+    expected_units = ['meters', 'feet']
+
+    folder.AddFromConfig([doc], '{0}/file.yaml'.format(_GOOD_PATH))
+    distance_units = folder.local_namespace.GetUnitsForMeasurement('distance')
+    length_units = folder.local_namespace.GetUnitsForMeasurement('length')
+
+    self.assertEmpty(folder.local_namespace.GetFindings())
+    self.assertCountEqual(expected_units, distance_units)
+    self.assertCountEqual(expected_units, length_units)
+
+  def testUnitFolderAddFromConfigBadAliasBase(self):
+    doc = {
+      'length': 'invalid',
+      'distance': [
+        {'meters': 'STANDARD'},
+        'feet'
+      ],
+    }
+    folder = unit_lib.UnitFolder(_GOOD_PATH)
+
+    folder.AddFromConfig([doc], '{0}/file.yaml'.format(_GOOD_PATH))
+
+    self.assertIsInstance(folder.local_namespace.GetFindings()[0],
+                          findings_lib.UnrecognizedMeasurementAliasBaseError)
+
+  def testUnitFolderAddFromConfigAliasIsAlias(self):
+    doc = {
+      'distance': [
+        {'meters': 'STANDARD'},
+        'feet'
+      ],
+      'length': 'distance',
+      'level': 'length',
+    }
+    folder = unit_lib.UnitFolder(_GOOD_PATH)
+
+    folder.AddFromConfig([doc], '{0}/file.yaml'.format(_GOOD_PATH))
+
+    self.assertIsInstance(folder.local_namespace.GetFindings()[0],
+                          findings_lib.MeasurementAliasIsAliasedError)
+
```

Comment:
add a test for DuplicateMeasurementAliasError?

```
    doc = {
      'distance': [
        {'meters': 'STANDARD'},
        'feet'
      ],
      'level': 'distance',
    }

    doc2 = {
      'length': [
        {'meters': 'STANDARD'},
        'feet'
      ],
      'level': 'length',
    }
```

Reply:
Added